AzureDevOpsPipelines/deployment/bot-publish/source/Core.ts

```diff
@@ -0,0 +1,110 @@
+import tl = require("azure-pipelines-task-lib");
+import { IExecSyncResult, IExecOptions } from 'azure-pipelines-task-lib/toolrunner';
+import { execSync } from "child_process";
+import path = require("path");
+import fs = require("fs");
+
+export class core {
+    private isLoggedIn: boolean = false;
+    private cliPasswordPath: string = "";
+    private servicePrincipalId: string = "";
+    private servicePrincipalKey: string = "";
+    private prefix: string = "";
+    
+    public subscriptionID: string = "";
+
+    constructor() {
+        this.getPrefix();
+        this.subscriptionID = tl.getVariable('subscriptionID');
+        
+        if (this.subscriptionID == null) {
+            this.loginAzureRM(tl.getInput("connectedServiceNameARM", true));
+            tl.setVariable('subscriptionID', this.subscriptionID);
+        }
+    }
+
+    private loginAzureRM(connectedService: string): void {
+        var authScheme: string = tl.getEndpointAuthorizationScheme(connectedService, true);
+        this.subscriptionID = tl.getEndpointDataParameter(connectedService, "SubscriptionID", true);
+        if(authScheme.toLowerCase() == "serviceprincipal") {
+            let authType: string = tl.getEndpointAuthorizationParameter(connectedService, 'authenticationType', true);
+            let cliPassword: string;
+            var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
+            if (authType == "spnCertificate") {
+                tl.debug('certificate based endpoint');
+                let certificateContent: string = tl.getEndpointAuthorizationParameter(connectedService, "servicePrincipalCertificate", false);
```

Comment:
Variable declarations'd come first. Also, if it's value is only set here, it'd be a const

Reply:
In this case it makes senes to log the debug line before the declaration of the variable, as it is calling `getEndpointAuthorizationParameter` method. If anything goes wrong with the authorization it should be logged as part of the `certificate based endpoint` log.

Reply:
Ok, perfect! But I think it'd be a const instead of let.